packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_counter/blocs/counter_bloc.dart

```diff
@@ -37,37 +43,54 @@ abstract class CounterBlocStates {
 
   /// Error messages
   Stream<String> get errors;
+
+  /// Result state
+  Stream<Result<Count>> get counterResult;
 }
 
 /// A BloC responsible for count calculations
 @RxBloc()
 class CounterBloc extends $CounterBloc {
-  /// Default constructor
-  CounterBloc(this._repository);
+  /// Bloc constructor
+  CounterBloc(this._repository) {
+    Rx.merge<Result<Count>>([
+      // On increment.
+      _$incrementEvent
+          .switchMap((_) => _repository.increment().asResultStream()),
+      // On decrement.
+      _$decrementEvent
+          .switchMap((_) => _repository.decrement().asResultStream()),
+      // Get current value
+      _$currentEvent
+          .switchMap((_) => _repository.getCurrent().asResultStream()),
+      // Get initial value
+      _repository.getCurrent().asResultStream(),
+    ]).bind(_lastFetchedCount).disposedBy(_compositeSubscription);
+  }
 
   final CounterRepository _repository;
+  final _lastFetchedCount = BehaviorSubject<Result<Count>>();
```

Comment:
No.
The `counterResult` is an already defined state with the return type of `Stream<Result<Count>>`. Your code in the constructor  (relating to the `Rx.merge`) can be moved so that it is passed to `counterResult`. Additionally, you can map this value to the `count`. This way, you'd remove the need for creating, listening and disposing of a behavior subject.

Reply:
I tried in this way but each time I hit plus or minus button, four requests to the server are sent. Maybe I map other states to the counterResult in wrong way.